images/router/haproxy/conf/haproxy-config.template

```diff
@@ -459,7 +465,11 @@ backend be_secure_{{$cfgIdx}}
   balance {{ $balanceAlgo }}
       {{ end }}
     {{ else }}
+      {{ if (matchPattern "roundrobin|leastconn|source" (env "ROUTER_LOAD_BALANCE_ALGORITHM" "")) }}
+  balance {{ env "ROUTER_LOAD_BALANCE_ALGORITHM" "leastconn"}}
```

Comment:
We don't expose everything in the oadm.  They can do 'oc set env dc/router ROUTER_LOAD_BALANCE_ALGORITHM=source' if they want to change it.

We need to make sure that we document the new environment variable and annotation.  And we should provide a snippet showing how to set ENV and annotations from the cli (with a link to the full docs).